lib/active_campaign/api/fields.rb

```diff
@@ -0,0 +1,162 @@
+      # @option params [String] :descript description of the field being
+      # @option params [true, false] :isrequired dictates whether the field being created is required or not
+      # @option params [String] :perstag the perstag that represents the field being created
+      # @option params [String] :defval the default value of the field being created
+      # @option params [true, false] :show_in_list show this field in the contact list view (deprecated - no longer used)
+      # @option params [Integer] :rows the number of rows for a textarea (deprecated - no longer used)
+      # @option params [Integer] :cols the number of columns for a textarea (deprecated - no longer used)
+      # @option params [true, false] :visible show or hide this field when using the Form Builder
+      # @option params [String] :service possible values: nimble, contactually, mindbody, salesforce, highrise,
+      #   google_spreadsheets, pipedrive, onepage, google_contacts, freshbooks, shopify, zendesk, etsy, NULL,
+      #   bigcommerce, capsule, bigcommerce_oauth, sugarcrm, zohocrm, batchbook
+      # @option params [Integer] :ordernum order of appearance in 'My Fields' tab
+      #
+      # @return [Hash] a hash with information about all custom fields, including the newly created custom field
+      def create_field(params)
+        post('fields', field: params)
+      end
+
+      #
+      # Retrieve a custom field
+      #
+      # @param [Integer] id the id of the custom field to show
+      #
+      # @return [Hash]
+      def show_field(id)
+        get("fields/#{id}")
+      end
+
+      #
+      # Update a custom field
+      #
+      # @param [Integer] id the id of the field to update
+      # @param [Hash] params update the custom field with this data
+      # @option params [String] :title the title of the field being created
+      # @option params [String] :type possible values: dropdown, hidden, checkbox, date, text, textarea, NULL, listbox, radio
+      # @option params [String] :descript description of the field being
+      # @option params [true, false] :isrequired dictates whether the field being created is required or not
+      # @option params [String] :perstag the perstag that represents the field being created
+      # @option params [String] :defval the default value of the field being created
+      # @option params [true, false] :show_in_list show this field in the contact list view (deprecated - no longer used)
+      # @option params [Integer] :rows the number of rows for a textarea (deprecated - no longer used)
+      # @option params [Integer] :cols the number of columns for a textarea (deprecated - no longer used)
+      # @option params [true, false] :visible show or hide this field when using the Form Builder
+      # @option params [String] :service possible values: nimble, contactually, mindbody, salesforce, highrise,
```

Comment:
Good idea with breaking the long lines like this. I'll see about doing that in other places later.

Reply:
I like to keep rubocop happy, haha